src/Compatibility/Microsoft.DotNet.PackageValidation/ValidatePackage.cs

```diff
@@ -30,30 +33,58 @@ public class ValidatePackage : TaskBase
 
         public string CompatibilitySuppressionFilePath { get; set; }
 
+        public string RoslynDirectory { get; set; }
+
         protected override void ExecuteCore()
         {
-            RuntimeGraph runtimeGraph = null;
-            if (!string.IsNullOrEmpty(RuntimeGraph))
+            if (string.IsNullOrEmpty(RoslynDirectory))
```

Comment:
Instead of placing this in `ExecuteCore` can we place it in an override of `Execute`?  That way we get this as close to the start of the task as possible.  It's also beneficial to put it in a method body that does little else, that way it reduces the chance that the JIT might go and load assemblies as part of compiling the code where we hook assembly resolution.

Reply:
how does this look ?
```c#
        public override bool Execute()
        {
            if (string.IsNullOrEmpty(RoslynDirectory))
            {
                throw new ArgumentNullException(nameof(RoslynDirectory));
            }

            AppDomain.CurrentDomain.AssemblyResolve += ResolverForRoslyn;
            try
            {
                base.Execute();
            }
            finally
            {
                AppDomain.CurrentDomain.AssemblyResolve -= ResolverForRoslyn;
            }

            return !Log.HasLoggedErrors;
        }
```

Reply:
Can you move the `return` statement into the `try` as `return base.Execute()`?